src/reducers/importText.ts

```diff
@@ -157,22 +160,55 @@ const importText = (state: State, { path, text, lastUpdated, preventSetCursor, r
 
   }
   else {
+
     const json = isRoam ? roamJsonToBlocks(JSON.parse(convertedText)) : convertHTMLtoJSON(convertedText)
-    const imported = importJSON(state, simplePath, json, { lastUpdated, skipRoot })
 
-    return reducerFlow([
+    const uuid = createId()
 
-      updateThoughts(imported),
+    // Note: Create a dummy thought and then import new thoughts into its context and then collpase it . Since collapse uses existingThoughtMove it merges imported thoughts.
+    const updatedState = newThought(state, {
+      at: simplePath,
+      insertNewSubthought: true,
+      value: uuid
+    })
 
-      // restore the selection to the last imported thought on the first level
-      !preventSetCursor
-        ? setCursor({
-          path: imported.lastImported,
-          offset: headValue(imported.lastImported).length
-        })
-        : null,
+    const destEmpty = destThought.value === '' && getAllChildren(state, pathToContext(simplePath)).length === 0
 
-    ])(state)
+    // New dummy thoughts for collapsing
+    const newDummyThought = getAllChildren(updatedState, pathToContext(simplePath)).find(child => child.value === uuid)
+
+    const newDestinationPath = unroot(newDummyThought ? simplePath.concat(newDummyThought) : simplePath)
```

Comment:
It's preferable to cast `newDestinationPath` to `SimplePath` as soon as it is assigned, as it is always a `SimplePath`.

Reply:
I suggest `[...simplePath, newDummyThought]` instead of `simplePath.concat(newDummyThought)`. It's clean and slightly faster. While I'm sure the performance is not relevant here, I think it's a good habit to get into since `Array.prototype.concat` doesn't offer any advantages based on my research.

Reply:
> I suggest `[...simplePath, newDummyThought]` instead of `simplePath.concat(newDummyThought)`. It's clean and slightly faster. While I'm sure the performance is not relevant here, I think it's a good habit to get into since `Array.prototype.concat` doesn't offer any advantages based on my research.

Thanks. I was not aware of the slight performance benefit.